### PR TITLE
add proxy support, make httpClient instance variable [#15]

### PR DESCRIPTION
**Pull request by dnl-blkv**

### CODE REVIEW

**Comment by dnl-blkv** [comment]
@OGKevin Had to make HttpClient an instance variable for a case if they want to use two different contexts with two different proxies within one process :)

File: BunqSdk/Http/ApiClient.cs [hunk 1]
```diff
@@ -49,15 +49,42 @@ public class ApiClient
...
+        {
+            return new HttpClient(CreateHttpClientHandler())
+            {
+                BaseAddress = new Uri(apiContext.GetBaseUri())
+            };
+        }
+
+        private HttpClientHandler CreateHttpClientHandler()
+        {
+            // TODO: Add HTTP Public Key Pinning. It is needed to prevent possible man-in-the-middle attacks using
```
Comment (OGKevin) [inline BunqSdk/Http/ApiClient.cs]: Is this todo supposed to be here ? If so could you remove it and create an issue for it with a link to this particular file and line ?

File: BunqSdk/Http/BunqProxy.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        public ICredentials Credentials { get; set; }
+
+        public Uri GetProxy(Uri destination)
+        {
+            return ProxyUri;
+        }
+
+        public bool IsBypassed(Uri host)
+        {
+            return false; /* Proxy all requests */
```
Comment (OGKevin) [inline BunqSdk/Http/BunqProxy.cs]: Method doc 🤔 ?

File: BunqSdk/Http/BunqProxy.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+using System.Net;
+
+namespace Bunq.Sdk.Http
+{
+    public class BunqProxy : IWebProxy
+    {
+        public BunqProxy(string proxyUri)
+            : this(new Uri(proxyUri))
+        {
+        }
```
Comment (OGKevin) [inline BunqSdk/Http/BunqProxy.cs]: Is there really a need for 2 lines for this 😝 can this be reformatted ?

Comment (dnl-blkv) [inline BunqSdk/Http/BunqProxy.cs]: @OGKevin this is in accordance with C# coding standard

File: BunqSdk/Http/ApiClient.cs (hunk 1 above)
Comment (dnl-blkv) [inline BunqSdk/Http/ApiClient.cs]: This TODO is supposed to be there, and we would like to keep it there as TODO. Issue can be created additionally (you can do it if you wish :P).

File: BunqSdk/Http/BunqProxy.cs (hunk 2 above)
Comment (dnl-blkv) [inline BunqSdk/Http/BunqProxy.cs]: Nope, because this refers to the documentation. `false` means "Nothing is bypassed" which means "proxy all requests".

**Comment by OGKevin** [comment]
@andrederoos

**Pull request merged**